include/swift/AST/Types.h

```diff
@@ -3035,8 +3083,21 @@ class AnyFunctionType : public TypeBase {
 
   GenericSignature getOptGenericSignature() const;
   
+  bool hasClangFunctionType() const {
+    return Bits.AnyFunctionType.HasUncommonInfo;
+  }
+
+  const clang::Type *getClangFunctionType() const;
+  const clang::Type *getCanonicalClangFunctionType() const;
+
   ExtInfo getExtInfo() const {
-    return ExtInfo(Bits.AnyFunctionType.ExtInfo);
+    return ExtInfo(Bits.AnyFunctionType.ExtInfoBits, getClangFunctionType());
+  }
+
+  ExtInfo getCanonicalExtInfo(bool useClangFunctionType) const {
```

Comment:
Please include a comment explaining that this parameter is only for staging purposes while we teach Swift to provide function types reliably.